modules/candidate_list/jsx/candidateListIndex.js

```diff
@@ -0,0 +1,363 @@
+          name: 'entityType',
+          type: 'select',
+          options: {
+            'Human': 'Human',
+            'Scanner': 'Scanner',
+          },
+        },
+      },
+      {
+        'label': 'Scan Done',
+        'show': true,
+        'filter': {
+          name: 'scanDone',
+          type: 'select',
+          hide: this.state.hideFilter,
+          options: {
+            'Y': 'Yes',
+            'N': 'No',
+          },
+        },
+      },
+      {
+        'label': 'Participant Status',
+        'show': true,
+        'filter': {
+          name: 'participantStatus',
+          type: 'select',
+          hide: this.state.hideFilter,
+          options: options.participantstatus,
+        },
+      },
+      {
+        'label': 'DoB',
+        'show': true,
+        'filter': {
+          name: 'DoB',
+          type: 'date',
+          hide: this.state.hideFilter,
+        },
+      },
+      {
+        label: 'Sex',
+        show: true,
+        filter: {
+          name: 'sex',
+          type: 'select',
+          hide: this.state.hideFilter,
+          options: {
+            'Male': 'Male',
+            'Female': 'Female',
+          },
+        },
+      },
+      {
+        'label': 'VisitCount',
+        'show': true,
+        'filter': {
+          name: 'visitCount',
+          type: 'text',
+          hide: this.state.hideFilter,
+        },
+      },
+      {
+        'label': 'Feedback',
+        'show': true,
+        'filter': {
+          name: 'feedback',
+          type: 'select',
+          hide: this.state.hideFilter,
+          options: {
+            '0': 'None',
+            '1': 'opened',
+            '2': 'answered',
+            '3': 'closed',
+            '4': 'comment',
+          },
+        },
+      },
+      {
+        'label': 'Latest Visit Status',
+        'show': true,
+        'filter': {
+          name: 'latestVisitStatus',
+          type: 'select',
+          hide: this.state.hideFilter,
+          options: {
+            'Not Started': 'Not Started',
+            'Screening': 'Screening',
+            'Visit': 'Visit',
+            'Approval': 'Approval',
+            'Recycling Bin': 'Recycling Bin',
+          },
+        },
+      },
+
+    ];
+
+    if (options.useedc === 'true') {
+      fields.push(
+        {
+          'label': 'EDC',
+          'show': true,
+          'filter': {
+            name: 'edc',
+            type: 'date',
+            hide: this.state.hideFilter,
+          },
+        }
+      );
+    }
+
+    // Basic/Advanced toggle
+    const toggleFilters = () => {
+      const newFilterState = this.state.hideFilter === true ? false : true;
+      let newLabel = '';
+      if (this.state.buttonLabel === 'Show Advanced Filters') {
+        newLabel = 'Hide Advanced Filters';
+      } else {
+        newLabel = 'Show Advanced Filters';
+      }
+      this.setState({hideFilter: newFilterState});
+      this.setState({buttonLabel: newLabel});
+    };
```

Comment:
I would write this as a function outside of the render function because it doesn't need to be rendered on every mount. It can also be condensed into:

```
toggleFilters() {
  const hideFilter = !this.state.hideFilter;
  const buttonLabel = hideFilter ? 'Show Advanced Filters' : 'Hide Advanced Filters';
  this.setState({hideFilter, buttonLabel});
}
```

Reply:
Actually, for some reason I cannot move it out of the render function. I get many errors if I do. At least I simplified it in the next commit.

Reply:
ha! I had an epiphany last night and realized why I got errors when moving the function out of the render() function. I forgot to bind the function in the constructor, duh... 

Anyway, all done!